core/src/main/java/eu/maveniverse/maven/njord/shared/impl/publisher/DefaultArtifactPublisherRedirector.java

```diff
@@ -72,22 +81,22 @@ private String getRedirectUrl(Map<String, String> config, RepositoryMode mode, R
             throw new IllegalStateException("Unknown repository mode: " + mode);
         }
         // if conf comes from server/config, and contains it unprefixed, use it
-        if (config.containsKey(SessionConfig.SERVER_ID_KEY) && config.containsKey(key)) {
-            return config.get(key);
+        if (serverConfig.containsKey(SessionConfig.SERVER_ID_KEY) && serverConfig.containsKey(key)) {
+            return serverConfig.get(key);
```

Comment:
this needs an additional condition to make sure to obey the order of precedence!

Reply:
Fixed in https://github.com/maveniverse/njord/pull/220/commits/21e2401dfde4c3f03ff4a0aca1e138bfae8f41b5